priv/signs.json

```diff
@@ -6019,7 +6019,7 @@
         {
           "stop_id": "70205",
           "direction_id": 1,
-          "headway_direction_name": "Lechmere",
+          "headway_direction_name": "Union Square",
           "routes": ["Green-B", "Green-C", "Green-D", "Green-E"],
```

Comment:
@losvedir if I could ask you a question here: Should we keep `"Green-B", "Green-C", "Green-D"` as part of the `routes` list since North Station is still part of green trunk? Or should we remove those for this sign since only E-line trains are going to be heading eastbound from here. Not sure if we would want alerts affecting this sign for branches that terminate at North Station. Although, I noticed Green-B is in here while the B Line doesn't go to North Station, so maybe we just include them all for any green trunk stop?

Reply:
The `routes` list serves two purposes.

The first is with alerts, as you've alluded to. But I don't think a `Green-B` alert would affect the sign out here, unless its stop ID was included for some reason in the alert (if that's even possible). I think it _might_ show up if it was an entire _route_ suspension, which actually might be bad because it would say "No train service", even though technically there could still be Union Sq <-> Park St service in that case.

However, that field is also [used for filtering predictions](https://github.com/mbta/realtime_signs/pull/166). Back when I did that PR, Kenmore station had two different berths that trains stopped at (one for Green-B and one for Green-C/Green-D) , with corresponding countdown signs for each, but only one underlying GTFS stop ID. So to filter the predictions that appeared on the sign we used a combination of the stop ID and the route, to get the right train predictions to the right sign. However, since then we actually created different stop IDs for the different berths, and you can see here in `signs.json` that one sign uses `71150` and the other `70150`. So we could probably omit the filtering functionality now. I can't think of any other instances on the subway where we have 1) different countdown signs requiring different boarding information and 2) the same underlying stop ID. Stations like Downtown Crossing with both Red and Orange service have different stop IDs, as does Park St with its different berths in both the eastbound and westbound directions.

But unless or until you remove the filtering functionality, you probably want to include all the routes here. It's _possible_ that occasionally you'll have Green-B/C/D predictions out here. It's more likely inbound, as a train could be told to go to Boston College / Cleveland Circle / Riverside, and the operator will key in their new AVI code, and we'll start predicting them to go out that way. An outbound train could possibly have a B/C/D route keyed in either through operator error or because they're planning to go out to Union Sq and then back to B/C/D branches.

Reply:
For instance, over the last two weeks, you can see we've [made a handful of predictions](https://prediction-analyzer.mbtace.com/accuracy?filters[bin]=All&filters[chart_range]=Daily&filters[date_end]=2022-03-16&filters[date_start]=2022-03-02&filters[direction_id]=any&filters[in_next_two]=&filters[mode]=subway&filters[route_ids]=Green-B&filters[service_date]=2022-03-15&filters[stop_ids][]=70204) for Green-B trains at Haymarket.

Reply:
Updating on this: I had a separate PR that removed the other routes from the config for Science Park, but I think it's best to just leave them since I think it'll affect the station in the same way as mentioned by Gabe above. Meaning - alerts from other lines probably won't affect Science Park unless for some reason the stop_id was included in the alert and in the slight chance that an inbound train at Science Park gets told to go to a different branch, those predictions would show on the Science Park signs.